### ISSUE TITLE
regression: Android Auto, crash parsing geo-intent

### ISSUE THREAD

**Issue by ntruchsess** [osmandapp/OsmAnd#19125]
### Description

PR #19115 attempted to fix issue #19038 but introduced a new crash parsing the intent.
crash happens in [String query = uri.getQueryParameter("q");](https://github.com/osmandapp/OsmAnd/blob/4c75209b4a4eb9749d79d16c458c76d28951e448/OsmAnd/src/net/osmand/plus/auto/NavigationSession.java#L250). The class android.net.Uri doesn't support parsing query-parameters of geo-urls. The parsing must be done in own code.

### Steps to reproduce

same steps as issue #19038

### Actual result

application crashes.

### Expected result

SearchResultScreen would open with coordinates respectivly searchphrase or address from intent. Unique search-result would trigger calculation of route and directly open NavigationScreen.

### Your Environment (required)

```
OsmAnd Version: 4.7.0 (build master commit 81267a773d913abfa7e856ea0344b865bfe46de7)
Android/iOS version: 12
Device model: Samsung Galaxy S10 (SG973F)
Crash-Logs: 
2024-02-16 18:38:53.213  3491-3491  AndroidRuntime          net.osmand.plus                      E  FATAL EXCEPTION: main
                                                                                                    Process: net.osmand.plus, PID: 3491
                                                                                                    java.lang.RuntimeException: java.lang.UnsupportedOperationException: This isn't a hierarchical URI.
                                                                                                    	at androidx.car.app.utils.RemoteUtils.lambda$dispatchCallFromHost$0(RemoteUtils.java:155)
                                                                                                    	at androidx.car.app.utils.RemoteUtils$$ExternalSyntheticLambda1.run(Unknown Source:6)
                                                                                                    	at androidx.car.app.utils.ThreadUtils.runOnMain(ThreadUtils.java:39)
                                                                                                    	at androidx.car.app.utils.RemoteUtils.dispatchCallFromHost(RemoteUtils.java:147)
                                                                                                    	at androidx.car.app.utils.RemoteUtils.lambda$dispatchCallFromHost$1(RemoteUtils.java:186)
                                                                                                    	at androidx.car.app.utils.RemoteUtils$$ExternalSyntheticLambda2.run(Unknown Source:8)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:938)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:226)
                                                                                                    	at android.os.Looper.loop(Looper.java:313)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8751)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
                                                                                                    Caused by: java.lang.UnsupportedOperationException: This isn't a hierarchical URI.
                                                                                                    	at android.net.Uri.getQueryParameter(Uri.java:1720)
                                                                                                    	at net.osmand.plus.auto.NavigationSession.onNewIntent(NavigationSession.java:250)
                                                                                                    	at androidx.car.app.CarAppService$CarAppBinder.onNewIntentInternal(CarAppService.java:610)
                                                                                                    	at androidx.car.app.CarAppService$CarAppBinder.lambda$onNewIntent$5$androidx-car-app-CarAppService$CarAppBinder(CarAppService.java:505)
                                                                                                    	at androidx.car.app.CarAppService$CarAppBinder$$ExternalSyntheticLambda3.dispatch(Unknown Source:6)
                                                                                                    	at androidx.car.app.utils.RemoteUtils.lambda$dispatchCallFromHost$0(RemoteUtils.java:150)
                                                                                                    	at androidx.car.app.utils.RemoteUtils$$ExternalSyntheticLambda1.run(Unknown Source:6) 
                                                                                                    	at androidx.car.app.utils.ThreadUtils.runOnMain(ThreadUtils.java:39) 
                                                                                                    	at androidx.car.app.utils.RemoteUtils.dispatchCallFromHost(RemoteUtils.java:147) 
                                                                                                    	at androidx.car.app.utils.RemoteUtils.lambda$dispatchCallFromHost$1(RemoteUtils.java:186) 
                                                                                                    	at androidx.car.app.utils.RemoteUtils$$ExternalSyntheticLambda2.run(Unknown Source:8) 
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:938) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:226) 
                                                                                                    	at android.os.Looper.loop(Looper.java:313) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8751) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135) 
```

**Comment by ntruchsess** [comment]
this issue is fixed by PR #19126

**Comment by ntruchsess** [comment]
crash fixed by commit 95878b058a5f9b18ce77fcd6b78d45e50b185cbb (parsing is still not working though, but that should be handled in #19130)
